IGSIOCalibration/vtkIGSIOAbstractCalibrationAlgo.h

```diff
@@ -0,0 +1,224 @@
+{
+public:
+  vtkTypeMacro(vtkIGSIOAbstractCalibrationAlgo, vtkObject);
+
+  enum Events
+  {
+    InputTransformAddedEvent = vtkCommand::UserEvent + 173,
+  };
+
+  enum CalibrationErrorCodes
+  {
+    CALIBRATION_FAIL = IGSIO_FAIL,
+    CALIBRATION_SUCCESS = IGSIO_SUCCESS,
+    CALIBRATION_NOT_STARTED,
+    CALIBRATION_NOT_ENOUGH_POINTS,
+    CALIBRATION_NOT_ENOUGH_VARIATION,
+    CALIBRATION_HIGH_ERROR,
+  };
+
+  /// Read configuration
+  /// \param aConfig Root element of the device set configuration
+  virtual igsioStatus ReadConfiguration(vtkXMLDataElement* aConfig) = 0;
+
+  /// Remove all previously inserted calibration points.
+  /// Call this method to get rid of previously added calibration points
+  /// before starting a new calibration.
+  virtual void RemoveAllCalibrationPoints();
+
+  /// Insert acquired point to calibration point list
+  /// \param aMarkerToReferenceTransformMatrix New calibration point (tool to reference transform)
+  virtual igsioStatus InsertNextCalibrationPoint(vtkMatrix4x4* aMarkerToReferenceTransformMatrix);
+
+  /// Get the number of outlier points. It is recommended to display a warning to the user
+  /// if the percentage of outliers vs total number of points is larger than a few percent.
+  virtual int GetNumberOfDetectedOutliers();
+
+  /// Get the total number of collected calibration points
+  virtual int GetNumberOfCalibrationPoints();
+
+  // Computes the maximum orientation difference in degrees between the first tool transformation
+  // and all the others. Used for determining if there was enough variation in the input data.
+  virtual double GetMaximumToolOrientationDifferenceDeg();
+
+  /// Name of the object marker coordinate frame (eg. Stylus)
+  vtkGetStringMacro(ObjectMarkerCoordinateFrame);
+  vtkSetStringMacro(ObjectMarkerCoordinateFrame);
+
+  /// Name of the reference coordinate frame (eg. Reference)
+  vtkGetStringMacro(ReferenceCoordinateFrame);
+  vtkSetStringMacro(ReferenceCoordinateFrame);
+
+  /// Name of the object pivot point coordinate frame (eg. StylusTip)
+  vtkGetStringMacro(ObjectPivotPointCoordinateFrame);
+  vtkSetStringMacro(ObjectPivotPointCoordinateFrame);
+
+  /// Error code indicating what went wrong with the calibration.
+  vtkGetMacro(ErrorCode, int);
+
+  //@{
+  /// Required minimum amount of variation within the recorded poses
+  vtkSetMacro(MinimumOrientationDifferenceDeg, double);
+  vtkGetMacro(MinimumOrientationDifferenceDeg, double);
+  //@}
+
+  //@{
+  /// Required minimum amount of variation in position from the previous position in order for a transform to be accepted (0 degrees by default).
+  vtkGetMacro(PositionDifferenceThresholdMm, double);
+  vtkSetMacro(PositionDifferenceThresholdMm, double);
+  //@}
+
+  //@{
+  /// Required minimum amount of variation in position from the previous position in order for a transform to be accepted (0 degrees by default).
+  vtkGetMacro(OrientationDifferenceThresholdDegrees, double);
```

Comment:
Decide how to write degrees and use it consistently (above we just used `deg`, here it is `degrees`). Maybe `deg` is sufficient.